list.py

```diff
@@ -137,11 +138,9 @@ def check_index_and_transform(self, index, include_border = 0):
             returns transformed index
         '''
 
-        if index > self._length + include_border or index < -self._length:
```

Comment:
It looks like I found a bug that breaks the program.

In case of an empty List and an attempt to [remove](https://github.com/nitkach/data_structures/pull/5/commits/ba25f8b0308a8bd0c347e717618de527fceffef9#diff-8ff389d9f192b8783e5093ac8c9c5ef34c6e298a115abad116528f0149d8b616L227) an element by 0 index, the exception will not rise, and the program will break ([`None.next`](https://github.com/nitkach/data_structures/pull/5/commits/ba25f8b0308a8bd0c347e717618de527fceffef9#diff-8ff389d9f192b8783e5093ac8c9c5ef34c6e298a115abad116528f0149d8b616L231) will be taken):
```Python
l = List()
l.remove(0)
```
```Python
if 0 > 0 + 0 or 0 < 0: # False
   ...
return 0
```
[L230](https://github.com/nitkach/data_structures/pull/5/commits/ba25f8b0308a8bd0c347e717618de527fceffef9#diff-8ff389d9f192b8783e5093ac8c9c5ef34c6e298a115abad116528f0149d8b616L230)
`self.head` points to `None` because the list is empty:
```Python
if 0 == 0: # True
   self.head = None.next

Reply:
Correct. The empty list is generally a special case in this implementation, and we need to specially handle it, and be careful about it.

There exists another approach for the implementation that involves using a sentinel node. I don't think it's important, and current approach is also good, but if you are interested here is an wiki article about sentinel nodes and their benefits (the link is anchored to a python example with double-linked list): [the link](https://en.wikipedia.org/wiki/Sentinel_node#Python_implementation_of_a_circular_doubly-linked_list)

Reply:
Regarding the markdown syntax of the comments in Github. In general, it's a good practice to put the links in markdown inside the text using `[text](link)` syntax. But Github has its own "flavor" of markdown with additional features like [permanent code links with a preview](https://docs.github.com/en/get-started/writing-on-github/working-with-advanced-formatting/creating-a-permanent-link-to-a-code-snippet).

For example, I could reference a block of code from your commit by just pasting a bare link like this:

https://github.com/nitkach/data_structures/blob/ba25f8b0308a8bd0c347e717618de527fceffef9/list.py#L1-L4

To do that I opened your git branch (in regular repository folders and files page, not in this PR diff view), and clicked "Copy permalink"
![image](https://user-images.githubusercontent.com/36276403/214757675-4b35d556-9fe1-4c60-ba7d-90130b3e2417.png)

The "perma" part of the "permalink" means "permanent". It means the link will always point to a specific commit with the state of code that was in that commit. If you ever decide to rewrite your code or even completely remove this Node class that I referenced in `master` branch, then this permalink in this comment won't be broken, it will always direct you to a specific commit and it doesn't care about what's in master